llvm-spirv/lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -2344,6 +2344,29 @@ static void transMetadataDecorations(Metadata *MD, SPIRVEntry *Target) {
                               SPIRVWord);
       TWO_INT_DECORATION_CASE(MathOpDSPModeINTEL, spv::internal, SPIRVWord,
                               SPIRVWord);
+	case DecorationConduitKernelArgumentINTEL:
+	case DecorationRegisterMapKernelArgumentINTEL:
+	case DecorationStableKernelArgumentINTEL:
+	case DecorationRestrict: {
+        Target->addDecorate(new SPIRVDecorate(DecoKind, Target));
+		break;
+	}
+	case DecorationBufferLocationINTEL:
```

Comment:
Is this expected to be here?

Reply:
Yes.  There is also buffer location in the kernel arg annotation and thus we reuse this value